src/rpc/blockchain.cpp

```diff
@@ -786,6 +786,49 @@ static RPCHelpMan getmempoolentry()
     };
 }
 
+static RPCHelpMan getmempoolspendingtx()
+{
+    return RPCHelpMan{"getmempoolspendingtx",
+                "\nScans the mempool to find a transaction spending the given output\n",
+                {
+                    {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "transaction id"},
+                    {"n", RPCArg::Type::NUM, RPCArg::Optional::NO, "vout number"},
+                },
```

Comment:
I'm wondering if, in the future, you might want to pass in a list of prevouts? (e.g. if you have transaction(s) with multiple inputs and want to see if anything in the mempool conflicts)

Reply:
That's a good idea, it's worth adding this flexibility right now to avoid running in backwards-compatibility issues if we add it later, I'll try that. I'll take a list of `prevouts` and return a map of `prevout -> spending tx (or null if not found)`.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/24408/commits/4ef5a16c306bb4784fb645d1df51fe0efeadcb06
I chose to return only the `spendingtxid` instead of the mempool tx data, because otherwise if a transaction appears multiple times (because it spends many of the provided prevouts) it would be duplicated, which is a (small) waste.

Callers can chain this with `getmempoolentry` to get more data about the conflicting transaction.
If you think it makes more sense to directly include the mempool tx let me know, I don't have strong feelings about that.